src/KeyringController.test.ts

```diff
@@ -122,7 +122,7 @@ describe('KeyringController', () => {
       await keyringController.persistAllKeyrings();
 
       const { vault } = keyringController.store.getState();
-      const keyrings = await mockEncryptor.decrypt(PASSWORD, vault);
+      const keyrings = await mockEncryptor.decrypt(PASSWORD, vault as string);
```

Comment:
Ah, I see that the `vault` is not set initially, so there's no way to remove the `undefined` from its type.

This comes up in the NetworkController tests (`provider` is similarly not set initially and set is later on) and the way I've solved is to assert it in the test if I need to do something with it. So what you do think about something like:

``` suggestion
      assert(vault, "Vault is not set");
      const keyrings = await mockEncryptor.decrypt(PASSWORD, vault);
```